closed/src/java.base/share/native/libjncrypto/NativeCrypto.c

```diff
@@ -1626,6 +1714,59 @@ JNIEXPORT jint JNICALL Java_jdk_crypto_jniprovider_NativeCrypto_RSAEP
     return (jint)msg_len;
 }
 
+/* RSADP Cryptographic Primitive, RSA Private Key operation
+ * Returns -1 on error
+ * The param verify is -1 for 'no verify', otherwise it is size of m (with verify)
+ *
+ * Class:     jdk_crypto_jniprovider_NativeCrypto
+ * Method:    RSADP
+ * Signature: ([BI[BIJ)I
+ */
+
+typedef struct RSADP_BN {
+    BN_CTX *ctx;
+    BIGNUM *k_bigNum;
+    BIGNUM *m_bigNum;
+    BIGNUM *rem;
+} RSADP_BN;
+
+/*
+ * Class:     jdk_crypto_jniprovider_NativeCrypto
+ * Method:    create_rsadp_BN
+ * Signature: ()J
+ */
+JNIEXPORT jlong JNICALL Java_jdk_crypto_jniprovider_NativeCrypto_create_1rsadp_1BN
+	(JNIEnv *env, jclass thisObj) {
+
+    RSADP_BN *rsadp_bn_ptr = (RSADP_BN *)malloc(sizeof(RSADP_BN));
+    if(rsadp_bn_ptr != NULL) {
+        rsadp_bn_ptr->ctx = (*OSSL_BN_CTX_new)();
```

Comment:
I expect these allocations can fail so need to be checked as well.

Reply:
If any allocation fails, the resources acquired via successful calls must be released.